File: scratch/check.py
import urllib.request
import json

url = 'https://firestore.googleapis.com/v1/projects/arisan-antigravity/databases/(default)/documents/groups'
try:
    with urllib.request.urlopen(url) as response:
        data = json.loads(response.read().decode())
        groups = data.get('documents', [])
        for group in groups:
            fields = group.get('fields', {})
            name = fields.get('name', {}).get('stringValue', '')
            if 'tretan' in name.lower():
                print(f"Group: {name}")
                members = fields.get('members', {}).get('arrayValue', {}).get('values', [])
                for m in members:
                    m_fields = m.get('mapValue', {}).get('fields', {})
                    waNumber = m_fields.get('waNumber', {}).get('stringValue', '')
                    userId = m_fields.get('userId', {}).get('stringValue', '')
                    role = m_fields.get('role', {}).get('stringValue', 'Member')
                    memberName = m_fields.get('name', {}).get('stringValue', '')
                    
                    print(f"- Member {memberName} ({waNumber} / userId:{userId}) is: {role}")
except Exception as e:
    print(e)

File: scratch/check.js
const https = require('https');

https.get('https://firestore.googleapis.com/v1/projects/arisan-antigravity/databases/(default)/documents/groups', (res) => {
  let data = '';
  res.on('data', chunk => data += chunk);
  res.on('end', () => {
    const json = JSON.parse(data);
    const groups = json.documents || [];
    let found = false;
    groups.forEach(group => {
      const name = group.fields?.name?.stringValue;
      if (name && name.toLowerCase().includes('tretan')) {
        console.log(`Found Group: ${name}`);
        const members = group.fields?.members?.arrayValue?.values || [];
        members.forEach(m => {
          const fields = m.mapValue?.fields;
          const waNumber = fields?.waNumber?.stringValue || '';
          const userId = fields?.userId?.stringValue || '';
          const role = fields?.role?.stringValue || 'Member';
          const memberName = fields?.name?.stringValue || '';
          
          if (waNumber.includes('badrusalsyihab') || userId.includes('badrusalsyihab')) {
            console.log(`- Member ${memberName} (${waNumber} / ${userId}) is: ${role}`);
            found = true;
          }
        });
      }
    });
    if (!found) console.log('Did not find the user in the tretan group.');
  });
});
